casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala

```diff
@@ -422,35 +431,27 @@ object ProtoUtil {
   def stringToByteString(string: String): ByteString =
     ByteString.copyFrom(Base16.decode(string))
 
-  def basicDeployData(id: Int): DeployData = {
-    //TODO this should be removed once we assign the deploy with exact user
-    Thread.sleep(1)
-    val timestamp = System.currentTimeMillis()
-    val term      = s"@${id}!($id)"
-
-    DeployData()
-      .withUser(ByteString.EMPTY)
-      .withTimestamp(timestamp)
-      .withTerm(term)
-      .withPhloLimit(accounting.MAX_VALUE)
-  }
-
-  def basicDeploy(id: Int): Deploy = {
-    val d    = basicDeployData(id)
-    val term = InterpreterUtil.mkTerm(d.term).right.get
-    Deploy(
-      term = Some(term),
-      raw = Some(d)
-    )
-  }
+  def basicDeployData[F[_]: Monad: Time](id: Int): F[DeployData] =
+    for {
+      now  <- Time[F].currentMillis
+      term = s"@${id}!($id)"
+    } yield
+      DeployData()
+        .withUser(ByteString.EMPTY)
+        .withTimestamp(now)
+        .withTerm(term)
+        .withPhloLimit(accounting.MAX_VALUE)
+
+  def basicDeploy[F[_]: Monad: Time](id: Int): F[Deploy] =
+    for {
+      d    <- basicDeployData[F](id)
+      term = InterpreterUtil.mkTerm(d.term).right.get
```

Comment:
I think this can fail and `.right.get` will throw.
```scala
      term <- Sync[F].fromEither(InterpreterUtil.mkTerm(d.term))
```
is cleaner and safer.